test/services/iothub/iothubClient-tests.js

```diff
@@ -0,0 +1,292 @@
+            client = new IoTHubClient(suite.credentials, suite.subscriptionId, null);
+            subscriptionId = suite.subscriptionId;
+        });
+
+        if (suite.isPlayback) {
+            client.longRunningOperationRetryTimeout = 0;
+        }
+
+        suite.createResourcegroup(resourceGroupName, location, function (err, result) {
+            should.not.exist(err);
+        });
+
+        done();
+    });
+
+    after(function (done) {
+        suite.teardownSuite(done);
+    });
+
+    beforeEach(function (done) {
+        suite.setupTest(done);
+    });
+
+    afterEach(function (done) {
+        suite.baseTeardownTest(done);
+    });
+
+
+    describe('IoTHub Lifecycle Test', function () {
+        it('should check if the iothub name exists', function (done) {
+
+            var operationInputs = {
+                name: resourceName
+            };
+
+            client.iotHubResource.checkNameAvailability(operationInputs, function (err, result, request, response) {
+                should.not.exist(err);
+                if (!result.nameAvailable) {
+                    iotHubExists = true;
+                    console.log('Hub exists');
+                } else {
+                    console.log('Hub does not exist');
+                }
+                done();
+            });
+        });
+
+        it('should delete a hub if it exists successfully', function (done) {
+
+            if (iotHubExists) {
+                client.iotHubResource.deleteMethod(resourceGroupName, resourceName, null, function (err, result, request, response) {
+                    should.not.exist(err);
+                    done();
+                });
+            } else {
+                console.log('Hub does not exist. Skipping Delete');
+                done();
+            }
+        });
+
+        it('should create the iothub successfully', function (done) {
+
+            var iotHubCreateParams = {
+                name: resourceName,
+                location: location,
+                subscriptionid: subscriptionId,
+                resourcegroup: resourceGroupName,
+                sku: {
+                    name: 'S1',
+                    capacity: 2
+                },
+                properties: {
+                    enableFileUploadNotifications: false,
+                    operationsMonitoringProperties: {
+                        events: {
+                            "C2DCommands": "Error",
+                            "DeviceTelemetry": "Error",
+                            "DeviceIdentityOperations": "Error",
+                            "Connections": "Error, Information"
+                        }
+                    },
+                    "features": "None",
+                }
+            }
+
+            console.log('Creating the hub');
+            client.iotHubResource.createOrUpdate(resourceGroupName, resourceName, iotHubCreateParams, null, function (err, result, request, response) {
```

Comment:
It is not required to pass null for options. The generated code will shift the callback by 1 to the left if the options object was not provided.